tools/clang/unittests/HLSL/DxilContainerTest.cpp

```diff
@@ -1762,6 +1763,274 @@ TEST_F(DxilContainerTest, DxcUtils_CreateReflection) {
   }
 }
 
+TEST_F(DxilContainerTest, CheckReflectionQueryInterface) {
+  if (m_ver.SkipDxilVersion(1, 5))
+    return;
+
+  // Check that QueryInterface for shader and library reflection accepts/rejects
+  // interfaces properly
+
+  // Also check that DxilShaderReflection::Get*ParameterDesc methods do not
+  // write out-of-bounds for earlier interface version.
+
+  // Use domain shader to test DxilShaderReflection::Get*ParameterDesc because
+  // it can use all three signatures.
+  const char *shaderSource = R"(
+struct PSInput {
+	noperspective float4 pos : SV_POSITION;
+};
+// Patch constant signature
+struct HS_CONSTANT_DATA_OUTPUT {
+	float edges[3] : SV_TessFactor;
+	float inside : SV_InsideTessFactor;
+};
+// Patch input signature
+struct HS_CONTROL_POINT {
+  float3 pos : POSITION;
+};
+// Domain shader
+[domain("tri")]
+void main(
+    OutputPatch<HS_CONTROL_POINT, 3> TrianglePatch,
+    HS_CONSTANT_DATA_OUTPUT pcIn,
+    float3 bary : SV_DomainLocation,
+    out PSInput output) {
+  output.pos = float4((bary.x * TrianglePatch[0].pos +
+                       bary.y * TrianglePatch[1].pos +
+                       bary.z * TrianglePatch[2].pos), 1);
+}
+  )";
+
+  CComPtr<IDxcUtils> pUtils;
+  VERIFY_SUCCEEDED(m_dllSupport.CreateInstance(CLSID_DxcUtils, &pUtils));
+
+  CComPtr<IDxcCompiler> pCompiler;
+  VERIFY_SUCCEEDED(CreateCompiler(&pCompiler));
+
+  struct TestDescStruct {
+    D3D12_SIGNATURE_PARAMETER_DESC paramDesc;
+    uint32_t pad;
```

Comment:
Understanding the code below makes it clear why this pad is needed, but I think it would be helpful to add a comment here.

Reply:
Will do, thanks.